deepecho/sequences.py

```diff
@@ -107,7 +107,8 @@ def segment_sequence(sequence, segment_size, sequence_index, drop_sequence_index
 def _convert_to_dicts(segments, context_columns):
     sequences = []
     for segment in segments:
-        if context_columns:
+        missing_columns = [col for col in context_columns if col not in segment.columns]
+        if context_columns and not missing_columns:
```

Comment:
I'm not sure we want to skip the entire check if there are missing columns present, ideally we'd want to run it with whatever context is present.